libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -266,36 +279,45 @@ class MapboxRouteLineApi(
-        return if (offset >= 0) {
-            val trafficLineExpression = MapboxRouteLineUtils.getTrafficLineExpression(
-                offset,
-                routeLineExpressionData,
-                routeLineOptions.resourceProvider.routeLineColorResources.routeUnknownTrafficColor
-            )
-            val routeLineExpression = MapboxRouteLineUtils.getVanishingRouteLineExpression(
-                offset,
-                routeLineOptions.resourceProvider.routeLineColorResources.routeLineTraveledColor,
-                routeLineOptions.resourceProvider.routeLineColorResources.routeDefaultColor
-            )
-            val routeLineCasingExpression =
-                MapboxRouteLineUtils.getVanishingRouteLineExpression(
+    fun setVanishingOffset(offset: Double):
+        Expected<VanishingRouteLineUpdateValue, RouteLineNoUpdate> {
+            routeLineOptions.vanishingRouteLine?.vanishPointOffset = offset
+            return if (offset >= 0) {
+                val trafficLineExpression = MapboxRouteLineUtils.getTrafficLineExpression(
                     offset,
+                    routeLineExpressionData,
                     routeLineOptions
-                        .resourceProvider.routeLineColorResources.routeLineTraveledCasingColor,
-                    routeLineOptions.resourceProvider.routeLineColorResources.routeCasingColor
+                        .resourceProvider
+                        .routeLineColorResources
+                        .routeUnknownTrafficColor
                 )
+                val routeLineExpression = MapboxRouteLineUtils.getVanishingRouteLineExpression(
+                    offset,
+                    routeLineOptions
+                        .resourceProvider
+                        .routeLineColorResources
+                        .routeLineTraveledColor,
+                    routeLineOptions.resourceProvider.routeLineColorResources.routeDefaultColor
+                )
+                val routeLineCasingExpression =
+                    MapboxRouteLineUtils.getVanishingRouteLineExpression(
+                        offset,
+                        routeLineOptions
+                            .resourceProvider.routeLineColorResources.routeLineTraveledCasingColor,
+                        routeLineOptions.resourceProvider.routeLineColorResources.routeCasingColor
+                    )
 
-            RouteLineState.VanishingRouteLineUpdateState(
-                trafficLineExpression,
-                routeLineExpression,
-                routeLineCasingExpression
-            )
-        } else {
-            null
+                Expected.Success(
+                    VanishingRouteLineUpdateValue(
+                        trafficLineExpression,
+                        routeLineExpression,
+                        routeLineCasingExpression
+                    )
+                )
+            } else {
+                Expected.Failure(RouteLineNoUpdate())
```

Comment:
```suggestion
                Expected.Failure(VanishingRouteLineUpdateError("offset too small to produce a meaningfull update, skipping"))
```

Or something along those lines.